packages/tlinks/src/ui/FarcasterFrame.tsx

```diff
@@ -0,0 +1,186 @@
+"use client";
+import {
+    useState,
+    useMemo,
+    useCallback,
+} from "react";
+import {
+    http,
+} from "wagmi";
+import { useFarcasterIdentity } from "@frames.js/render/identity/farcaster";
+import { useFrame } from "@frames.js/render/use-frame";
+import {
+    fallbackFrameContext,
+    type OnTransactionArgs,
+    signFrameAction,
+} from "@frames.js/render";
+import {
+    FrameUI,
+    type FrameUIComponents,
+} from "@frames.js/render/ui";
+import { WebStorage } from "@frames.js/render/identity/storage";
+import { useNeynarContext } from "@neynar/react";
+import { tokenType } from "@token-kit/onchain";
+import { createPublicClient, type PublicClient } from "viem";
+import { getBlockExplorerUrl, getRPCURL, theme, type FarcasterUser, type StylingProps } from "../utils/constants.ts";
+interface FarcasterFrameProps {
+    chainId: number;
+    scriptURI?: string;
+    cssClass?: string;
+    farcaster: FarcasterUser
+}
+
+const VIEWER_PATTERNS = [
+    "viewer",
+    "viewer-staging",
+    "https://ipfs.io/ipfs",
+    "ipfs://ipfs",
+    "ipfs://",
+];
+
+export async function isViewer(
+    chainId: number,
+    contract: `0x${string}`,
+) {
+    const client = createPublicClient({
+        transport: http(getRPCURL(chainId)),
```

Comment:
Does this work when you tested it? these API request will get CORS error

Reply:
changed to using common api.